### PR TITLE
add builtin fun round_real

### PR DESCRIPTION
**Pull request by laidahe**
## What have you changed? (mandatory)

add builtin fun round_real

## What are the type of the changes? (mandatory)

Improvement 

## How has this PR been tested? (mandatory)

unit test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

no

## Does this PR affect tidb-ansible update? (mandatory)

no

## Refer to a related PR or issue link (optional)

[3275](https://github.com/tikv/tikv/issues/3275)

### CODE REVIEW

**Comment by hicqu** [comment]
LGTM.

**Comment by hicqu** [comment]
Please fix the test.

File: src/coprocessor/dag/expr/builtin_math.rs [hunk 1]
```diff
@@ -251,6 +258,45 @@ mod test {
...
+                Datum::F64(-3.45),
+                Datum::F64(-3f64),
+            ),
+            (
+                ScalarFuncSig::RouldReal,
+                Datum::F64(f64::MAX),
+                Datum::F64(f64::MAX),
+            ),
+            (
+                ScalarFuncSig::RouldReal,
```
Comment (AndreMouche) [inline src/coprocessor/dag/expr/builtin_math.rs]: Could you help update the [tipb](https://github.com/pingcap/tipb ) to the newest version to rename RouldRead to RoundReal?

Comment (laidahe) [reply src/coprocessor/dag/expr/builtin_math.rs]: ok

Comment (laidahe) [reply src/coprocessor/dag/expr/builtin_math.rs]: i found that these is an error showing "no method named `set_encode_type` found for type `tipb::select::StreamResponse` " when compiling tikv after tipb updated.

Comment (AndreMouche) [reply src/coprocessor/dag/expr/builtin_math.rs]: got it, I'll update it later then.

**Comment by AndreMouche** [comment]
Please fix the CI and `DCO`

File: src/coprocessor/dag/expr/builtin_math.rs (hunk 1 above)
Comment (laidahe) [reply src/coprocessor/dag/expr/builtin_math.rs]: When u have updated,tell me then.

**Comment by huachaohuang** [comment]
@breeswish PTAL

File: src/coprocessor/dag/expr/scalar_function.rs [hunk 2]
```diff
@@ -400,6 +401,7 @@ impl ScalarFunc {
             | ScalarFuncSig::Right
             | ScalarFuncSig::RightBinary
             | ScalarFuncSig::RouldReal
+            | ScalarFuncSig::RightShift
```
Comment (breeswish) [inline src/coprocessor/dag/expr/scalar_function.rs]: Why is this line added?

**Comment by huachaohuang** [comment]
@laidahe Would you like to address the comments?

**Comment by zhangjinpeng1987** [comment]
@laidahe Can you sign-off and fix the conflict?

**Comment by breeswish** [comment]
RoundReal has been implemented in this PR: #3621 

I'm going to close this PR.
